raphtory-graphql/src/model/graph/namespace.rs

```diff
@@ -0,0 +1,109 @@
+use crate::{
+    data::get_relative_path,
+    model::graph::meta_graph::MetaGraph,
+    paths::{valid_path, ExistingGraphFolder},
+};
+use dynamic_graphql::{ResolvedObject, ResolvedObjectFields};
+use std::path::PathBuf;
+use walkdir::WalkDir;
+
+#[derive(ResolvedObject)]
+pub(crate) struct Namespace {
+    base_dir: PathBuf,
+    current_dir: PathBuf,
+}
+
+impl Namespace {
+    pub fn new(base_dir: PathBuf, current_dir: PathBuf) -> Self {
+        Self {
+            base_dir,
+            current_dir,
+        }
+    }
+
+    fn get_all_graph_folders(&self) -> Vec<ExistingGraphFolder> {
+        let base_path = self.base_dir.clone();
+        WalkDir::new(&self.current_dir)
+            .max_depth(1)
+            .into_iter()
+            .filter_map(|e| {
+                let entry = e.ok()?;
+                let path = entry.path();
+                let relative = get_relative_path(base_path.clone(), path).ok()?;
+                let relative_str = relative.to_str()?; // potential UTF8 error here
+                let cleaned = relative_str.replace(r"\", "/");
+                let folder = ExistingGraphFolder::try_from(base_path.clone(), &cleaned).ok()?;
+                Some(folder)
+            })
+            .collect()
+    }
+
+    pub(crate) fn get_all_children(&self) -> Vec<Namespace> {
+        WalkDir::new(&self.current_dir)
+            .into_iter()
+            .filter_map(|e| {
+                let entry = e.ok()?;
+                let file_name = entry.file_name();
+                let file_name = file_name.to_string_lossy();
+                let file_name = file_name.to_string();
+                let file_name = file_name.as_str();
+                let path = entry.path();
+                if path.is_dir()
+                    && path != self.current_dir
+                    && valid_path(self.current_dir.clone(), file_name, true).is_ok()
+                {
+                    Some(Namespace::new(self.base_dir.clone(), path.to_path_buf()))
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+}
+
+#[ResolvedObjectFields]
+impl Namespace {
+    async fn graphs(&self) -> Vec<MetaGraph> {
+        self.get_all_graph_folders()
+            .iter()
+            .map(|g| MetaGraph::new(g.clone()))
+            .collect()
+    }
+    async fn path(&self) -> Option<String> {
+        let relative = get_relative_path(self.base_dir.clone(), self.current_dir.as_path()).ok()?;
+        let relative_str = relative.to_str()?; // potential UTF8 error here
```

Comment:
we should never wrap something that is invalid UTF8?

Reply:
Simimlar to above this is taken from data.rs - it would only happen if the underlying folder created by the user has some funky non UTF8 characters in it - possible best these are just ignored anyway?